### PR TITLE
Added component docs

### PR DESCRIPTION
**Pull request by jlubean**
Added jsdoc comments for auto-documentation.

### CODE REVIEW

File: fs-anchored-dialog.html [hunk 1]
```diff
@@ -111,20 +107,120 @@
...
+        'transition':  {
+          type: string,
+          value: 'center'
+        },
+        /**
+         * Whether or not the modal should take up the entire screen.
+         * Default: true.
+         * @type {boolean} [fullscreen-on-mobile=true]
+         */
+        'fullscreen-on-mobile':  {
```
Comment (rosemariesadler) [inline fs-anchored-dialog.html]: Component design though: this attr seems counter intuitive - maybe we can rename it to something that would be easier to just set to true when we want to opt out. Any ideas @jlubean @crowtherdr @skye2k2?

File: fs-anchored-dialog.html [hunk 2]
```diff
@@ -111,20 +107,120 @@
...
+        'no-transition':  {
           type: Boolean,
           value: false
         }
       }
     }
+
+    /**
+     * Open the dialog.
+     * @param {Object} config - Object containing the following optional properties:
```
Comment (rosemariesadler) [inline fs-anchored-dialog.html]: just a small note - these aren't optional - focusBackElement is required and we need one of the other two.

File: fs-modal-dialog.html [hunk 3]
```diff
@@ -29,7 +29,102 @@
...
+     * The config object argument to pass to the public .open() function.
+     * @typedef {Object} ConfigObject
+     * @property {element} focusBackElement - The element to return focus back to after the dialog closes. Default: document.activeElement (at the time of opening the dialog).
+     */
+
+     /**
+      * `<fs-modal-dialog>` is a modal dialog centered on the screen that is displayed above other content, masks the content underneath it, and prevents the user from interacting with the content underneath.
+      *
+      *     <body>
+      *       <fs-modal-dialog></fs-modal-dialog>
```
Comment (rosemariesadler) [inline fs-modal-dialog.html]: we may want to flesh out these examples slightly more

File: fs-anchored-dialog.html (hunk 1 above)
Comment (skye2k2) [reply fs-anchored-dialog.html]: `no-fullscreen-mobile`

**Pull request merged**